CMakeLists.txt

```diff
@@ -0,0 +1,137 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.3)
+
+set(PACKAGE_NAME      "csi")
+set(PACKAGE_VERSION   "0.1.0")
+
+project(${PACKAGE_NAME})
+
+enable_language(CXX)
+
+set(CSI_SPEC    ${CMAKE_CURRENT_SOURCE_DIR}/spec.md)
+set(CSI_PROTO   ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}.proto)
+
+set(CSI_CXX_PROTO_HEADERS
+  ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}.pb.h
+  ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}.grpc.pb.h
+)
+
+set(CSI_CXX_PROTO_SOURCES
+  ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}.pb.cc
+  ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}.grpc.pb.cc
+)
+
+set(CSI_CXX_PROTO_GEN
+  ${CSI_CXX_PROTO_HEADERS}
+  ${CSI_CXX_PROTO_SOURCES}
+)
+
+#############################
+# Build rules.
+#############################
+
+set(CMAKE_CXX_STANDARD 11)
+
+find_program(PROTOBUF_COMPILER NAMES protoc)
```

Comment:
How about doing `find_program` only if `PROTOBUF_COMPILER` is not set? Then the user can set up it directly instead of setting up `CMAKE_PROGRAM_PATH`. Ditto below.